### PR TITLE
fix (DirectoryTree) : selectedNodes in onSelect method could not get a value when configuring fieldNames 

### LINKED ISSUE #44947: Antd Tree 修改fieldNames.key为其他值时，DirectoryTree 的onSelect方法中的selectedNodes无法正确获取值
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-p96w9l?file=/index.js)

### Steps to reproduce

将fieldNames.key改为'id'后，DirectoryTree无法通过id筛选出selectedNodes，将数据的id改回key就能够成功识别

### What is expected?

DirectoryTree的selectedNodes可以正确识别自定的key值

### What is actually happening?

返回了空数组[]

| Environment | Info |
| --- | --- |
| antd | 5.9.1 |
| React | ^18.2.0 |
| System | macOS 13.5.1 |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by Zian502**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #44947

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix (DirectoryTree) : selectedNodes in onSelect method could not get a value when configuring fieldNames    |
| 🇨🇳 Chinese |     修复(DirectoryTree)：配置fieldNames时，onSelect方法中的selectedNodes无法获取值      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 5852bba</samp>

Added a new feature to the `DirectoryTree` component that allows users to specify custom field names for the `treeData` prop. Updated the utility functions and added a test case to ensure the feature works correctly.

### <samp>🤖 Generated by Copilot at 5852bba</samp>

*  Add `fieldNames` prop to `DirectoryTree` component to allow customizing the field names of the `treeData` nodes ([link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-abd41216def304abcca63211e726233e8c9e736775dee6d0cd4df926d981ccdfL47-R47))
*  Pass `fieldNames` prop to `convertDirectoryKeysToNodes` function to respect custom field names when selecting nodes in `DirectoryTree` ([link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-abd41216def304abcca63211e726233e8c9e736775dee6d0cd4df926d981ccdfL140-R142), [link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-abd41216def304abcca63211e726233e8c9e736775dee6d0cd4df926d981ccdfL154-R158), [link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-abd41216def304abcca63211e726233e8c9e736775dee6d0cd4df926d981ccdfL160-R166))
*  Modify `traverseNodesKey` and `convertDirectoryKeysToNodes` functions in `components/tree/utils/dictUtil.ts` to accept and use `fieldNames` prop or default values ([link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-da0e832e3ec129967618fc6f25dc3004ee84def0fed079587ecebe1a146741dfL9-R36), [link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-da0e832e3ec129967618fc6f25dc3004ee84def0fed079587ecebe1a146741dfL74-R113))
*  Import `fillFieldNames` function and `TreeProps` type in `components/tree/utils/dictUtil.ts` to fill missing field names and type check `fieldNames` prop ([link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-da0e832e3ec129967618fc6f25dc3004ee84def0fed079587ecebe1a146741dfR2-R3))
*  Add test case to `components/tree/__tests__/directory.test.tsx` to check that `fieldNames` prop works as expected in `DirectoryTree` ([link](https://github.com/ant-design/ant-design/pull/45036/files?diff=unified&w=0#diff-1af094226c29ceaeedf3eaff5f6972884cac3c6f8783315a1ee56c1fda2ff92fR267-R293))

### CODE REVIEW

**Comment by Zian502** [comment]
大佬们帮忙 review 下

**Comment by yoyo837** [comment]
有相关issue可关联吗?

File: components/tree/DirectoryTree.tsx [hunk 1]
```diff
@@ -137,7 +137,9 @@ const DirectoryTree: React.ForwardRefRenderFunction<RcTree, DirectoryTreeProps>
       newSelectedKeys = keys;
       lastSelectedKey.current = key;
       cachedSelectedKeys.current = newSelectedKeys;
-      newEvent.selectedNodes = convertDirectoryKeysToNodes(treeData, newSelectedKeys);
+      newEvent.selectedNodes = convertDirectoryKeysToNodes(treeData, newSelectedKeys, {
```
Comment (yoyo837) [inline components/tree/DirectoryTree.tsx:140]: 三处字面量对象, 可以提取成一个吗?

**Review by zombieJ — APPROVED** [review]

**Comment by Zian502** [comment]
> 有相关issue可关联吗?

https://github.com/ant-design/ant-design/issues/44947

File: components/tree/DirectoryTree.tsx (hunk 1 above)
Comment (Zian502) [reply components/tree/DirectoryTree.tsx]: 建议第三个参数是以对象类型，考虑到后续的拓展性，还有这里有调用到 traverseNodesKey 函数，traverseNodesKey 函数的第三个参数目前被设计成配置形式

Comment (yoyo837) [reply components/tree/DirectoryTree.tsx]: 我的意思是提取变量, 不是不让放对象

Comment (Zian502) [reply components/tree/DirectoryTree.tsx]: ok,那我理解错了 ^ ^ !

**Pull request merged**